DependencyInjection/Compiler/EntityListenerPass.php

```diff
@@ -52,7 +52,20 @@ public function process(ContainerBuilder $container)
                     $this->attachToListener($container, $name, $id, $attributes);
                 }
 
-                $container->getDefinition($resolver)->addMethodCall('register', array(new Reference($id)));
+                if (isset($attributes['lazy']) && $attributes['lazy']) {
+                    $listener = $container->getDefinition($id);
+                    if (!$listener->isPublic()) {
+                        throw new \InvalidArgumentException(sprintf('The service "%s" must be public as this entity listener is lazy-loaded.', $id));
+                    }
+
+                    if ($listener->isAbstract()) {
+                        throw new \InvalidArgumentException(sprintf('The service "%s" must not be abstract as this entity listener is lazy-loaded.', $id));
+                    }
+
+                    $container->getDefinition($resolver)->addMethodCall('registerService', array($listener->getClass(), $id));
```

Comment:
This won't work if the user defined their own value for the `doctrine.orm.entity_listener_resolver.class` parameter that isn't your custom custom container aware entity resolver.